### PR TITLE
chore(core): Convert the MetricTags alias into a newtype wrapper

### PR DESCRIPTION
**Pull request by bruceg**
This change adds an opaque newtype wrapper for all uses of metric tags which
allows us to control the interface and semantics of the operations on this type.
This wrapper will be used to simplify support for alternate metric tags storage.

### CODE REVIEW

File: src/sinks/splunk_hec/metrics/encoder.rs [hunk 1]
```diff
@@ -55,8 +55,8 @@ impl HecMetricsEncoder {
             .into_iter()
             .flatten()
             // skip the metric tags used for templating
-            .filter(|(k, _)| !metadata.templated_field_keys.contains(k))
-            .map(|(k, v)| (k.as_str(), HecFieldValue::Str(v.as_str())))
+            .filter(|(k, _)| !metadata.templated_field_keys.iter().any(|f| f == k))
```
Comment (tobz) [inline src/sinks/splunk_hec/metrics/encoder.rs:58]: Could this not use `MetricTags::contains_key`?

**Review by tobz — APPROVED** [review]
Looks good to me, with one non-blocking nit. 👍🏻

**Review by lukesteensen — APPROVED** [review]

File: src/sinks/splunk_hec/metrics/encoder.rs (hunk 1 above)
Comment (bruceg) [reply src/sinks/splunk_hec/metrics/encoder.rs:58]: I'm not seeing how `MetricTags::contains_key` could be used here. What we want is a copy of all the tags that are not in `templated_field_keys`, which is type `Vec<String>`.

Comment (tobz) [reply src/sinks/splunk_hec/metrics/encoder.rs:58]: Ah derp, yeah... I was thinking it was doing something `contains_key`-esque, but now I see it was just doing the same exact thing as you have here. Ignore me. :)

**Pull request merged**